src/crimson/os/seastore/cache.h

```diff
@@ -1328,6 +1403,69 @@ class Cache {
     );
   }
 
+  template <typename T>
+  get_extent_ret<T> read_extent(
+    TCachedExtentRef<T>&& extent,
+    extent_len_t offset,
+    extent_len_t length
+  ) {
+    if (extent->is_ptr()) {
+      return read_extent(std::move(extent));
+    }
+    assert(extent->state == CachedExtent::extent_state_t::CLEAN_PENDING);
+    extent->set_io_wait();
+    auto old_length = extent->get_valid_length();
+    return seastar::do_with(
+      extent->read_buffer(offset, length),
+      [extent, this](auto &read_regions) {
+        return read_ertr::parallel_for_each(
+          read_regions,
+          [extent, this](auto &read_region) {
+            extent_len_t start = p2align(read_region.first,
+              epm.get_block_size());
+            extent_len_t end = p2roundup(
+              read_region.first + read_region.second,
+              epm.get_block_size());
+            extent_len_t aligned_length = end - start;
+            return seastar::do_with(
+              alloc_cache_buf(aligned_length),
```

Comment:
I think `alloc_cache_buf()` (i.e. allocating page aligned buffer) is not necessary for partial reads that are fragmented, block aligned should be fine and has better performance.